src/telliot_feeds/sources/wsteth_source.py

```diff
@@ -0,0 +1,96 @@
+from dataclasses import dataclass
+from dataclasses import field
+from typing import Any
+from typing import Optional
+
+from telliot_core.apps.telliot_config import TelliotConfig
+
+from telliot_feeds.dtypes.datapoint import OptionalDataPoint
+from telliot_feeds.pricing.price_service import WebPriceService
+from telliot_feeds.pricing.price_source import PriceSource
+from telliot_feeds.sources.price.spot.coingecko import CoinGeckoSpotPriceSource
+from telliot_feeds.sources.price.spot.coinpaprika import CoinpaprikaSpotPriceSource
+from telliot_feeds.sources.price.spot.curvefi import CurveFinanceSpotPriceSource
+from telliot_feeds.sources.price.spot.uniswapV3 import UniswapV3PriceSource
+from telliot_feeds.sources.price_aggregator import PriceAggregator
+from telliot_feeds.utils.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class WstETHSpotPriceService(WebPriceService):
+    """Custom WstETH Price Service"""
+
+    def __init__(self, **kwargs: Any) -> None:
+        kwargs["name"] = "Custom WstETH Price Service"
+        kwargs["url"] = ""
+        super().__init__(**kwargs)
+
+    def get_wsteth_steth_ratio(self) -> Optional[float]:
+        # get endpoint
+        cfg = TelliotConfig()
```

Comment:
Maybe move the instantiation of the config object to the init, rather than this function, so it's not done every time.